Updated test suite

saiHemak opened the pull request:
Updated test suite to support asynchronous execution

## Description
Updated the test suite to perform the check, drop, create databases in sequential manner.

## Motivation and Context
Encountered test failures due to asynchronous execution of the above said operations while testing with Kitura-NIO API.The reason being the difference in the time taken by ```HTTP.request```  is causing the test case to check for database existence even before creation 

## How Has This Been Tested?
Ran Kitura-CouchDB test suite

## Checklist:
- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

saiHemak committed:
Updated test suite

djones6 commented on Tests/CouchDBTests/CouchDBTest.swift:
```diff
@@ -52,62 +52,37 @@ class CouchDBTest: XCTestCase {
+    // setting up  the database
+    func setUpDatabase(isSetUpCompleted: @escaping () -> Void) -> Void {
+        self.dropDatabaseIfExists() {
+            self.createDatabase() {
```
`createDatabase()` also needs a delay here, so that we do not exhaust the Cloudant API rate limit (currently the `setUpDatabase()` function will perform 2 API calls in 1 second).

saiHemak replied:
Sure will do ..

djones6 commented on Tests/CouchDBTests/DBTests.swift:
```diff
@@ -36,25 +36,28 @@ class DBTests: CouchDBTest {
-                print(">> Database successfully deleted")
+                print(">> Database successfully deleted ")
+            }
+            self.couchDBClient.createDB(self.dbName) {(db: Database?, error: NSError?) in
```
Is this right? It looks like `createDB` and `deleteDB` will potentially be executed in parallel.  The delete used to be invoked during the completion for create, since those callbacks could (potentially) be invoked asynchronously.

djones6 reviewed (changes requested):
Please could you add the extra delay to `setupDatabase`.  In theory we should also have delays around the two calls in `testDB()`, however in practice we'll probably get away without them, because it's very unlikely multiple Travis builds would execute this test at the same second.

saiHemak commented:
@djones6  I have added delays in createDatabase and dropDatabase  methods. I have also modified ```DBTests.testDB```  to make calls in sequential manner

djones6 reviewed (approved):
This looks good now, thanks.
